EKF/ekf.cpp

```diff
@@ -235,8 +256,8 @@ bool Ekf::initialiseTilt()
 	const float accel_norm = _accel_lpf.getState().norm();
 	const float gyro_norm = _gyro_lpf.getState().norm();
 
-	if (accel_norm < 0.9f * CONSTANTS_ONE_G ||
-	    accel_norm > 1.1f * CONSTANTS_ONE_G ||
+	if (accel_norm < 0.8f * CONSTANTS_ONE_G ||
+	    accel_norm > 1.2f * CONSTANTS_ONE_G ||
```

Comment:
Did you really have an issue with that?

Reply:
Not personally, but a few times different users have reported problems/confusion that turned out to the the estimator not initializing due to how bad the lsm303d (original pixhawk, pixhawk 2, mindracer, etc) can be.

Reply:
To be clear, the solution in those cases was re-calibration, the problem is without the estimator running there's nothing that really indicates what's wrong or what a user should do.

Reply:
Why don't we use the gravity vector to check the orientations for accel calibration? It only needs to know which side it is on which doesn't require running an ahrs.

Reply:
> Why don't we use the gravity vector to check the orientations for accel calibration?

We do.
 - https://github.com/PX4/PX4-Autopilot/blob/149ac16bb442fc64eef320794a306b755758b5d4/src/modules/commander/accelerometer_calibration.cpp#L257-L308


The question here is more about the filter silently failing (not initializing) vs starting and then the system likely complaining about high accel bias or one of the preflight innovation failures. Alternatively we could think about adding something simpler between sensors and preflight checks that catch this and tell the user to re-calibrate (maybe move vehicle_at_reset out of ecl). This isn't a widespread problem, but it does come up.

Reply:
Opening this up to a range of 1.2 to 0.8 would allow up to acos(1/1.2) = 33 deg of initial tilt error due to to a horizontal acceleration at the moment of alignment. I will assess via replay what this does to the EKF, but a warning every 10 seconds that this check fails that says 'keep vehicle or check accel calibration' could be a better option.

Reply:
I ran a test with adding a 33.5 deg pitch offset to the ekf initialiseTilt function and the EKF handled it OK and both attitude and gyro bias states had stabilised within 20 seconds.

![Screen Shot 2020-12-16 at 5 42 00 pm](https://user-images.githubusercontent.com/3596952/102314158-172f9180-3fc6-11eb-8995-a0e597c6f37e.png)